File: backend/src/main/java/com/nishtha/RoomEase/property/service/PropertyService.java
package com.nishtha.RoomEase.property.service;

import com.nishtha.RoomEase.property.dto.*;
import com.nishtha.RoomEase.property.entity.Property;

import java.util.List;

public interface PropertyService {

    PropertyResponse createProperty(RegisterPropertyRequest request);

    List<PropertyResponse> getMyProperties();

    List<PropertyResponse> getAllAvailableProperties();

    PropertyResponse getProperty(Long id);

    Property getOwnerProperty(Long propertyId);

    PropertyResponse getPropertyDetails(Long id);

    PropertyResponse updateProperty(
            Long id,
            UpdatePropertyRequest request
    );

    void deleteProperty(Long id);

    List<PropertyResponse> searchProperties(PropertySearchRequest request);

    OwnerDashboardResponse getOwnerDashboard();

}

File: frontend/src/services/propertyService.js
import api from "./api";

const getMyProperties = () =>
    api.get("/properties");

const createProperty = (data) =>
    api.post("/properties/register-property", data);

const getProperty=(id)=>
    api.get(`/properties/${id}`);

const updateProperty = (id, data) =>
    api.put(`/properties/edit-property/${id}`, data);

const deleteProperty = (id) =>
    api.delete(`/properties/${id}`);

const getAllAvailableProperties = () => {
    return api.get("/properties/all");
};

const getPropertyDetails = (propertyId) => {
    return api.get(`/properties/details/${propertyId}`);
};

const searchProperties = async (searchRequest) => {
    console.log(searchRequest);
    const response = await api.post("/properties/search",searchRequest);
    console.log(response.data);
    return response.data;
};

const getOwnerDashboard=async ()=>{
    return api.get("properties/dashboard");
}

export default {
    getMyProperties,
    getProperty,
    createProperty,
    updateProperty,
    deleteProperty,
    getAllAvailableProperties,
    getPropertyDetails,
    searchProperties,
    getOwnerDashboard
};